src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.Visuals.cs

```diff
@@ -0,0 +1,402 @@
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpPrimaryButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "PrimaryAsDefaultButton";
+							}
+						}
+						else if (defaultButton == ContentDialogButton.Secondary)
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpSecondaryButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "SecondaryAsDefaultButton";
+							}
+						}
+						else if (defaultButton == ContentDialogButton.Close)
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpCloseButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "CloseAsDefaultButton";
+							}
+						}
+					}
+
+					GoToState(useTransitions, defaultButtonState);
+				}
+			}
+
+			{
+				// DialogShowingStates
+				if (m_placementMode == PlacementMode.InPlace)
+				{
+					GoToState(true, m_isShowing && !m_hideInProgress ? "DialogShowing" : "DialogHidden");
+				}
+				else if (m_placementMode != PlacementMode.Undetermined)
+				{
+					// For ContentDialog's shown in the popup, set the state to always showing since the opened
+					// state of the popup effectively controls whether its showing it not.
+					GoToState(false, "DialogShowingWithoutSmokeLayer");
+				}
+
+				// DialogSizingStates
+				{
+					GoToState(useTransitions, fullSizeDesired ? "FullDialogSizing" : "DefaultDialogSizing");
+				}
+
+				// DialogBorderStates
+				{
+					GoToState(useTransitions, "NoBorder");
+				}
+			}
+
+			// On PhoneBlue, the dialog did not move out of the way of the input pane.
+			if (true)
+			{
+				AdjustVisualStateForInputPane();
+			}
+		}
+
+		private void ResetAndPrepareContent()
+		{
+			// Uno TODO: understand when this is applicable (seemingly while dialog is open?)
+			//if (HasValidAppliedTemplate() && m_tpCurrentAsyncOperation && m_hasPreparedContent)
+			//{
+			//	m_hasPreparedContent = false;
+			//	PrepareContent();
+			//}
+		}
+
+		private void PrepareContent()
+		{
+			global::System.Diagnostics.Debug.Assert(HasValidAppliedTemplate());
+
+			if (!m_hasPreparedContent)
+			{
+				// Uno note: removed unused code (PlacementMode::TransplantedRootInPopup isn't used by most recent template)
+
+				UpdateVisualState();
+				UpdateTitleSpaceVisibility();
+
+				if (m_placementMode != PlacementMode.InPlace)
+				{
+					SizeAndPositionContentInPopup();
+				}
+
+				// Uno TODO (for now this would have to be applied in-app using ElevatedView from Uno.UI.Toolkit)
+				//// Cast a shadow
+				//(ApplyElevationEffect(m_tpBackgroundElementPart.AsOrNull<UIElement>()));
+
+				m_hasPreparedContent = true;
+			}
+		}
+
+		private void SizeAndPositionContentInPopup()
+		{
+			var m_tpPopup = _popup;
+
+			double xOffset = 0;
+			double yOffset = 0;
+
+			var windowBounds = Xaml.Window.Current.Bounds;
+
+			var flowDirection = FlowDirection;
+
+			FrameworkElement spBackgroundAsFE = m_tpBackgroundElementPart;
```

Comment:
Codacy found an issue: [Remove the unused local variable 'spBackgroundAsFE'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6389897)